plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/DirectoryLister.java

```diff
@@ -25,4 +26,8 @@
 {
     RemoteIterator<LocatedFileStatus> list(FileSystem fs, Table table, Path path)
             throws IOException;
+
+    default void invalidate(Partition partition) {}
```

Comment:
These methods are required because transaction level `CachingDirectoryLister` needs to delegate invalidation to global `CachingDirectoryLister`. I don't see a value in passing same object instance twice (one as `DirectoryLister` and another as `TableInvalidationCallback`). While it's not perfect to have `invalidation` methods in `DirectoryLister`, I don't think it's a big issue since `DirectoryLister` is really built around `CachingDirectoryLister`

Reply:
> We could have two delegates:

We could have that, but these would point to same object. Since there aren't really `TableInvalidationCallback` implementations that are unrelated to `DirectoryLister` and there is concept of `DirectoryLister` delegate now, I don't see a need for two interfaces. If it comes that we need separate `TableInvalidationCallback`, we can always restore that